packages/cli/src/ui/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,455 @@
+    setPendingSettings((prev) => ({
+      ...prev,
+      accessibility: { ...(prev.accessibility || {}), [key]: value },
+    }));
+    setShowRestartPrompt(true);
+  };
+  const updateCheckpointing = (
+    key: keyof CheckpointingSettings,
+    value: boolean,
+  ) => {
+    setPendingSettings((prev) => ({
+      ...prev,
+      checkpointing: { ...(prev.checkpointing || {}), [key]: value },
+    }));
+    setShowRestartPrompt(true);
+  };
+  const updateFileFiltering = (
+    key: keyof FileFilteringSettings,
+    value: boolean,
+  ) => {
+    setPendingSettings((prev) => ({
+      ...prev,
+      fileFiltering: { ...(prev.fileFiltering || {}), [key]: value },
+    }));
+    setShowRestartPrompt(true);
+  };
+  const updateSetting = <K extends keyof Settings>(
+    key: K,
+    value: Settings[K],
+  ) => {
+    setPendingSettings((prev) => ({ ...prev, [key]: value }));
+    setShowRestartPrompt(true);
+  };
+
+  // List of boolean settings (with default false if undefined)
+  const items = [
+    {
+      label: 'Show Memory Usage',
+      value: 'showMemoryUsage',
+      checked: getScopedValue('showMemoryUsage') ?? false,
+      toggle: () => {
+        updateSetting(
+          'showMemoryUsage',
+          !(getScopedValue('showMemoryUsage') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Disable Loading Phrases (Accessibility)',
+      value: 'accessibility.disableLoadingPhrases',
+      checked:
+        getScopedNestedValue('accessibility', 'disableLoadingPhrases') ?? false,
+      toggle: () => {
+        updateAccessibility(
+          'disableLoadingPhrases',
+          !(
+            getScopedNestedValue('accessibility', 'disableLoadingPhrases') ??
+            false
+          ),
+        );
+      },
+    },
+    {
+      label: 'Enable Usage Statistics',
+      value: 'usageStatisticsEnabled',
+      checked: getScopedValue('usageStatisticsEnabled') ?? false,
+      toggle: () => {
+        updateSetting(
+          'usageStatisticsEnabled',
+          !(getScopedValue('usageStatisticsEnabled') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Enable Checkpointing',
+      value: 'checkpointing.enabled',
+      checked: getScopedNestedValue('checkpointing', 'enabled') ?? false,
+      toggle: () => {
+        updateCheckpointing(
+          'enabled',
+          !(getScopedNestedValue('checkpointing', 'enabled') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Auto Configure Max Old Space Size',
+      value: 'autoConfigureMaxOldSpaceSize',
+      checked: getScopedValue('autoConfigureMaxOldSpaceSize') ?? false,
+      toggle: () => {
+        updateSetting(
+          'autoConfigureMaxOldSpaceSize',
+          !(getScopedValue('autoConfigureMaxOldSpaceSize') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Respect .gitignore',
+      value: 'fileFiltering.respectGitIgnore',
+      checked:
+        getScopedNestedValue('fileFiltering', 'respectGitIgnore') ?? false,
```

Comment:
these ignore options implicitly has a default value of true although it isn't obvious from the code... the default value isn't specified in settings where it would make sense but later when this is used.

I'd suggest we define these default values and nice human readable names in a json style object in settings.ts. That will set us up to export a json schema from them as well in the future.